notification-portlet-webapp/src/main/java/org/jasig/portlet/notice/service/jpa/JpaAddressee.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Licensed to Jasig under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Jasig licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.portlet.notice.service.jpa;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+/**
+ * Represents a named addressee for a notification.  There are two types of
+ * addressee:  individual and group.  The addressees list is the target(s) of
+ * the notification as the author selected them, e.g. "send this notification to
+ * graduating seniors."  The (aggregated) recipients list is the list of all
+ * users who received the notification, e.g. "johndoe," "janedoe," and
+ * "robsmith."  A single user <em>may</em> belong to two (or more) addressee
+ * groups for a single notification, but should only see <em>one copy</em> of
+ * the notification in such a case.
```

Comment:
Might be good to mention that adressee's are optional and used for tracking notifications sent to a group.